encoding/decoder_types.go

```diff
@@ -485,34 +485,56 @@ type mapAsStructDecoder struct {
 func (d *mapAsStructDecoder) decode(dv, sv reflect.Value) {
 	for _, kv := range sv.MapKeys() {
 		var f *field
+		var compoundFields = []*field{}
 		var fieldDec decoderFunc
 		key := []byte(kv.String())
 		for i := range d.fields {
 			ff := &d.fields[i]
 			ffd := d.fieldDecs[i]
+
 			if bytes.Equal(ff.nameBytes, key) {
 				f = ff
 				fieldDec = ffd
-				break
 			}
 			if f == nil && ff.equalFold(ff.nameBytes, key) {
 				f = ff
 				fieldDec = ffd
 			}
-		}
 
-		if f == nil {
-			continue
+			if f != nil && f.compound {
+				compoundFields = append(compoundFields, f)
+			} else if f != nil {
+				break
+			}
 		}
 
-		dElemVal := fieldByIndex(dv, f.index)
-		sElemVal := sv.MapIndex(kv)
+		if len(compoundFields) > 0 {
```

Comment:
I believe this says that a field in the DB can either be mapped to/from one field in the Go struct or to multiple fields using the compound key syntax, but you can mix the two. Which makes sense. Might be nice to state it explicitly in the docs?